src/ui/widgets/tabbox.cpp

```diff
@@ -0,0 +1,109 @@
+#include "tabbox.h"
+#include "ui/core.h"
+#include "ui/text.h"
+#include "ui/ui.h"
+#include "ui/boxgouraud.h"
+
+namespace ui::widgets
+{
+
+constexpr int TitleOffset = 10;
+constexpr int InnerMargin = 10;
+constexpr int WidgetOffsetTop = ui::FontHeight / 2 + InnerMargin + ui::OutlineBorderWidth;
+constexpr int TotalVerticalMargin = WidgetOffsetTop + InnerMargin + 2 * ui::OutlineBorderWidth;
+
+
+Tab::Tab(const std::string& title)
+    : m_title(std::make_unique<Checkbox>(title))
```

Comment:
I think the visual layout is too complex. maybe something like this works better - dimming inactive tabs, add a lot of spacing between headers, and a smaller rectangle next to the active tab. other ideas are welcome, too.

![grafik](https://user-images.githubusercontent.com/7853507/173202110-0f37bd25-3e96-4ca2-bcff-6657a75a4d01.png)

(positioning isn't accurate, it's more about the colors I wanted to show.)

Reply:
Maybe just make the header checkboxes borderless to remove some clutter.

Reply:
I'll adpot that suggestion, i'm having difficulty drawing dimmed text.

Reply:
looks good what you did.